legend-sdlc-server/src/main/java/org/finos/legend/sdlc/server/project/ProjectStructurePlatformExtensions.java

```diff
@@ -0,0 +1,160 @@
+    {
+        MutableMap<String, Platform> platformsMap = Maps.mutable.empty();
+        platforms.forEach(platform ->
+        {
+            if (platformsMap.containsKey(platform.name))
+            {
+                throw new IllegalArgumentException("Multiple platforms defined for platform '" + platform.name + "'");
+            }
+            platformsMap.put(platform.name, platform);
+        });
+        MutableMap<String, ExtensionsCollection> extensionsCollectionsMap = Maps.mutable.empty();
+        collections.forEach(collection ->
+        {
+            if (!platformsMap.containsKey(collection.platform))
+            {
+                throw new IllegalArgumentException("No platform metadata found for platform '" + collection.platform + "'");
+            }
+            if (extensionsCollectionsMap.containsKey(collection.name))
+            {
+                throw new IllegalArgumentException("Multiple extensions collection defined for extension '" + collection.name + "'");
+            }
+            extensionsCollectionsMap.put(collection.name, collection);
+        });
+        return new ProjectStructurePlatformExtensions(platformsMap, extensionsCollectionsMap);
+    }
+
+    public List<Platform> getPlatforms()
+    {
+        return this.platforms.values().stream().collect(Collectors.toList());
+    }
+
+    public Platform getPlatform(String platform)
+    {
+        if (!this.platforms.containsKey(platform))
+        {
+            throw new IllegalArgumentException("No platform metadata found for platform '" + platform + "'");
+        }
+        return this.platforms.get(platform);
+    }
+
+    public ExtensionsCollection getExtensionsCollection(String extension)
+    {
+        if (!this.containsExtension(extension))
+        {
+            throw new IllegalArgumentException("No extension collection found for extension name '" + extension + "'");
+        }
+        return this.extensionsCollections.get(extension);
+    }
+
+    public boolean containsExtension(String extension)
+    {
+        return this.extensionsCollections.containsKey(extension);
+    }
+
+    public static class Platform
+    {
+        private final String name;
+        private final String groupId;
+        private final Map<Integer, String> projectStructureVersions;
+
+        public Platform(String name, String groupId, Map<Integer, String> projectStructureVersionsMap)
+        {
+            this.name = name;
+            this.groupId = groupId;
+            this.projectStructureVersions = projectStructureVersionsMap;
+        }
+
+        public String getName()
+        {
+            return name;
+        }
+
+        public String getGroupId()
+        {
+            return groupId;
+        }
+
+        public Map<Integer, String> getProjectStructureVersions()
+        {
+            return projectStructureVersions;
+        }
```

Comment:
Do we need to give access to this map? If not, then remove this method. If we do, then let's make sure it's unmodifiable (e.g., by wrapping it with Collections.unmodifiableMap).